config/watcher.go

```diff
@@ -142,11 +142,22 @@ func newFSWatcher(ctx context.Context, configPath string, logger logging.Logger,
 			case <-cancelCtx.Done():
 				return
 			case event := <-fsWatcher.Events:
-				if event.Op&fsnotify.Write == fsnotify.Write {
+				// Monitor WRITE and REMOVE events.
+				// Editors that save in place WRITE over the monitored file.
+				// Editors that save atomically write to a temp file and swap. Events on original file are: RENAME->CHMOD->REMOVE
+				if event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Remove == fsnotify.Remove {
 					debounced(func() {
 						logger.Info("On-disk config file changed. Reloading the config file.")
 						//nolint:gosec
 						rd, err := os.ReadFile(configPath)
+						defer func() {
```

Comment:
I imagine the extra lines here are the linter complaining about error checks? Don't need to see a change here we have one way of working around this (when the function you care about needs parameters). It's not really much better:
```
robot/web/web.go:266:		defer utils.UncheckedErrorFunc(func() error { return os.RemoveAll(filepath.Dir(addr)) })
```

If the function you care about doesn't need parameters, this can be a better substitute:
```
robot/packages/utils.go:139:	defer utils.UncheckedErrorFunc(f.Close)
```